### PR TITLE
allow PaymentsDetailsResponse to deserialize responses that include an 'action' parameter

### PR DESCRIPTION
**Pull request by jrparker**
**Description**
While trying to update our SDK from version 7.1.0 to 18.0.0, I ran into an issue deserializing 3DS responses from the endpoint `https://checkout-test.adyen.com/v69/payments/details`

When I make a request to that endpoint similar to:
```json
{
  "details": {
    "threeds2.fingerprint": "FINGERPRINT_DATA..."
  },
  "paymentData": "PAYMENT_DATA...",
  "threeDSAuthenticationOnly": false
}
```

I receive a response similar to:
```json
{
  "resultCode": "ChallengeShopper",
  "action": {
    "paymentData": "PAYMENT_DATA...",
    "paymentMethodType": "scheme",
    "authorisationToken": "AUTHORISATION_TOKEN...",
    "subtype": "challenge",
    "token": "TOKEN...",
    "type": "threeDS2"
  }
}
```
It looks like `PaymentsResponse` and `PaymentsDetailsResponse` were previously updated to be two separate data models during the update to Checkout v67, but there is no `action` parameter in `PaymentsDetailsResponse`.
I do not see `action` as a response parameter in the [Checkout v69 documentation](https://docs.adyen.com/api-explorer/#/CheckoutService/v69/post/payments/details) for that endpoint either, but I'm receiving a response with the `action` parameter from that endpoint. The fact that this endpoint documentation does not match responses I am receiving is concerning as well.

The changes in this PR allow my 3DS tests to run successfully through our test account. I've moved `CheckoutPaymentsAction` and `ResultCodeEnum` from `PaymentsResponse` to `AbstractPaymentsResponse`, where they are included in both `PaymentsResponse` and `PaymentsDetailsResponse`.

**Tested scenarios**
I've added a unit test to ensure that the response I'm receiving from `https://checkout-test.adyen.com/v69/payments/details` can be deserialized to include the `action` data.

**Fixed issue**:

### CODE REVIEW

File: src/main/java/com/adyen/model/checkout/AbstractPaymentsResponse.java [hunk 1]
```diff
@@ -479,5 +503,70 @@ public Date getExpiryDate() {
...
+     * * **IdentifyShopper** – The issuer requires the shopper's device fingerprint before the payment can be authenticated. Returned for 3D Secure 2 transactions.
+     * * **Pending** – Indicates that it is not possible to obtain the final status of the payment. This can happen if the systems providing final status information for the payment are unavailable, or if the shopper needs to take further action to complete the payment. For more information on handling a pending payment, refer to [Payments with pending status](https://docs.adyen.com/development-resources/payments-with-pending-status).
+     * * **Received** – Indicates the payment has successfully been received by Adyen, and will be processed. This is the initial state for all payments.
+     * * **RedirectShopper** – Indicates the shopper should be redirected to an external web page or app to complete the authorisation.
+     * * **Refused** – Indicates the payment was refused. The reason is given in the &#x60;refusalReason&#x60; field. This is a final state.
+     */
+    @JsonAdapter(ResultCodeEnum.Adapter.class)
+    public enum ResultCodeEnum {
```
Comment (mjball) [inline src/main/java/com/adyen/model/checkout/AbstractPaymentsResponse.java:520]: I'm not super familiar with how this project organizes its classes, but there's enough in here it seems reasonable to me to move this enum to its own file.

**Comment by wboereboom** [comment]
Hi @jrparker,

Thanks for opening this PR.

All models in this repository are/will be auto-generated. So changing them will result in a temporary change that will be overridden once we update and generate all classes again.

I am currently in touch with our API devs to figure out why you are getting this action back from our API. Do you have a PSP reference for us to analyze?

Kind Regards,
Wouter
Adyen

**Comment by jrparker** [comment]
Hi @wboereboom,

A pspReference I have for this is `LLK44J4FF4BLNK82`. I am also able to see the request / response in our Adyen instance API Logs within the Adyen UI that shows the `action` parameter being returned for the call to the `/checkout/v69/payments/details` endpoint.

Please let me know if there is any other information you need.

**Comment by AlexandrosMor** [comment]
Hello  @jrparker,

Thank you for your contribution. We are planning to autogenerate the models so all the abstract classes in the models would be overwritten. The new models are going to be released soon. 

kind regards,
Alexandros
Adyen

**Pull request closed without merging**